src/components/SingleList.jsx

```diff
@@ -1,13 +1,19 @@
+import { useNavigate } from 'react-router-dom';
 import './SingleList.css';
 
-export function SingleList({ name, path, setListPath }) {
-	function handleClick() {
+export function SingleList({ name, path, setListPath, setLoading }) {
+	const navigate = useNavigate();
+
+	async function handleClick() {
+		setLoading(true);
+		await setListPath('');
```

Comment:
Question: is this line needed? I commented it out and didn't notice a difference in the UI.

Reply:
Sorry, are you referring to line 2 or a different line, @deeheber?

Reply:
No problem. I'm referring to line 9 the `await setListPath('');`

Mostly because on line 10 we're calling it again with a value instead of empty string. Felt redundant unless there's some extra context I'm missing.

Reply:
Thanks and yes, line 9 is needed. When we revisit the same shopping list page twice in a row, the value for "path" (the link for the respective shopping list) needs to be cleared and set again. Otherwise, the page will show a perpetual loading message. The state of our "loading" function (which we incorporated into the preexisting useShoppingListData function in firebase.js) won't change unless the path is cleared and then reset. This is why I called setListPath with an empty string before passing in the path on the next line. On a related note, the loading function was necessary to implement for other aspects of our code for this issue.

Reply:
@deeheber regarding your main comment, it would be great if we could create a new issue for the end of our project to address the state in the inputs not being cleared.

Reply:
Got it. Thanks for the details! Seems reasonable to me.

Reply:
No problem!